File: pagereplacement.c
#include<stdio.h>
int pagereplacement(int k, int pages[],int n)
{
      int frame[k];
      for(int i=0; i<k; i++)
      {
        frame[i] = -1;
      }
    int faults=0, hit=0;
    int len_frame =  k; //sizeof(frame)/sizeof(int);
    int len = len_frame;

    for(int i=0; i<n; i++)
    {
        int count = 0;
        for(int j=0; j<len_frame; j++)
        {
            if(pages[i] == frame[j])
            {
                count++;
            }
        }
        if(count != 0)
        {
            hit++;
        }
        else
          {
            if(len >= len_frame)
              {
                len = 0;
              }
              frame[len] = pages[i];
              len++;
              faults++;
          }
    }
    printf("no. of frames : %d      | faults : %d    hits   : %d\n", k, faults, hit);
//}
}
int main()
{
  int pages[] = {1,2,3,4,2,1,5,6,2,1,2,3,7,6,3,2,1,2,3,6};
  for(int k=1; k<8; k++)
  {
    pagereplacement(k, pages, sizeof(pages)/sizeof(int));
  }

}

File: pageReplacement.py
import random


def pageReplace(frame, pages, n):
    faults = 0
    hit = 0
    j = n + 1
    print("")

    for i in pages:
        if i not in frame:
            if (len(frame) < n):
                frame.append(i)
                faults = faults + 1
            else:
                if (j >= len(frame)):
                    j = 0
                frame[j] = i
                j = j + 1
                faults = faults + 1
        else:
            hit = hit + 1
    return (faults, hit)


if __name__ == '__main__':
    frame = []
    print("According to FIFO, the page which first comes in the memory will first goes out :\n")
    pages = [random.randint(0,9) for page in range(0,15)]
    #pages = [3, 2, 1, 0, 3, 2, 4, 3, 2, 1, 0, 4]
    print("\n\nPage References : ", pages)
    for i in range(1, 8):
        print("Number of frames : ", i)
        faults, hits = pageReplace(frame, pages, i)
        print("Faults : ",faults)
        print("Hits   : ", hits)
        print("\n------------------------")

